advanced-game-design-team/Artifacts/Raycast Collisions/README.md

```diff
@@ -0,0 +1,112 @@
+# Raycasted Point Collision
+
+![](./rays.gif)
+
+## Introduction
+
+This guide explores a 2D raycasting system designed for detecting collisions between rays emanating from a point (typically the mouse cursor) and rectangular obstacles. The system comprises several key components that work together to create an efficient collision detection mechanism.
+
+## Core Components
+
+### 1. Ray Generation
```

Comment:
The explanation in **1.** focuses on the use of `ray`s, but then immediately segues into using a vector of `ray_hit`s, rather than `ray`s. I think 1-2 lines explaining (high-level) how we go from "these are the rays we'll be using" to "but the main part we want is actually ray hits, not the rays themselves" would be helpful - even just hand-waving away something like "we'll get to that soon" would be enough if the change from rays to ray_hits is explicitly noted in the text.